src/microbe_stage/FluidCurrentDisplay.cs

```diff
@@ -0,0 +1,74 @@
+﻿using Godot;
+
+/// <summary>
+///   Displays fluid currents in the microbe stage
+/// </summary>
+public partial class FluidCurrentDisplay : GpuParticles3D
+{
+#pragma warning disable CA2213
+    private ShaderMaterial material = null!;
+
+    private MicrobeStage stage = null!;
+#pragma warning restore CA2213
+
+    private StringName gameTimeParameterName = new("gameTime");
+    private StringName speedParameterName = new("speed");
+    private StringName chaoticnessParameterName = new("chaoticness");
+    private StringName scaleParameterName = new("scale");
+
+    private float time;
+
+    public void Init(MicrobeStage microbeStage)
```

Comment:
Giving this the entire stage feels pretty wrong to me, so I think it would be better to just give a `IWorldSimulation worldTimeSource` as parameter.